File: src/main/java/my/study/repository/design/pattern/adapter/ObjectAdapter.java
package my.study.repository.design.pattern.adapter;

/**
 * @author djh on  2018/8/10 00:00
 * @E-Mail 1544579459@qq.com
 */
public class ObjectAdapter implements Dst {
    private Src mSrc;

    ObjectAdapter(Src src) {
        mSrc = src;
    }


    @Override
    public int mobileCharge() {
        if (mSrc != null) {
            System.out.println("适配器开始适配电压!");
            int adapterVoltage = mSrc.originalVoltage() / 44;
            System.out.println("适配后的电压:" + adapterVoltage);
            return adapterVoltage;
        }
        return 0;
    }
}

File: src/main/java/my/study/repository/design/pattern/adapter/kotlin/ObjectAdapter.kt
package my.study.repository.design.pattern.adapter.kotlin

/**
 * @author djh on  2018/10/4 16:12
 * @E-Mail 1544579459@qq.com
 */
class ObjectAdapter(
        private val src: Src
) : Dst {

    override fun mobileCharge(): Int {
        println("开始适配电压!")
        val voltage = src.standardVoltage() / 4
        println("适配后的电压: $voltage")
        return voltage
    }
}